modin/pandas/groupby.py

```diff
@@ -376,27 +377,22 @@ def aggregate(self, func=None, *args, **kwargs):
         relabeling_required = False
         if isinstance(func, dict) or func is None:
 
-            def _reconstruct_func(func, **kwargs):
-                relabeling_required, func, new_columns, order = reconstruct_func(
-                    func, **kwargs
-                )
-                # We convert to the string version of the function for simplicity.
-                func = {
-                    k: v
-                    if not callable(v) or v.__name__ not in dir(self)
-                    else v.__name__
-                    for k, v in func.items()
-                }
-                return relabeling_required, func, new_columns, order
-
-            relabeling_required, func_dict, new_columns, order = _reconstruct_func(
+            def try_get_str_func(o):
+                if not isinstance(o, str) and isinstance(o, Iterable):
+                    return [try_get_str_func(v) for v in o]
+                return o.__name__ if callable(o) and o.__name__ in dir(self) else o
+
+            relabeling_required, func_dict, new_columns, order = reconstruct_func(
                 func, **kwargs
             )
+            func_dict = {k: try_get_str_func(v) for k, v in func_dict.items()}
 
             if any(i not in self._df.columns for i in func_dict.keys()):
                 from pandas.core.base import SpecificationError
 
                 raise SpecificationError("nested renamer is not supported")
+            if func is None:
+                kwargs = {}
```

Comment:
Why `kwargs` should be empty?

Reply:
if `func is None`, then it means that aggregation function was specified via `kwargs`:
```python
grp.aggregate(max=("col1", "max"), new_sum=("col2", "sum"))
```
we pass `func` and `kwargs` to `reconstruct_func` that returns us a single dict with all aggregation functions, so we don't need `kwargs` anymore and it should be empty. Otherwise, `kwargs` will be passed to a backend like kwargs parameters for aggregation function, which is not true.